dynamodb/toddl.go

```diff
@@ -28,6 +28,19 @@ import (
 // Spanner. It uses the source schema in conv.SrcSchema, and writes
 // the Spanner schema to conv.SpSchema.
 func schemaToDDL(conv *internal.Conv) error {
+	// Tracks Spanner names that have been used for indexes. We use this to ensure we generate unique names when
+	// we map from DynamoDB to Spanner since we want Spanner table names and index names to be distinct.
+	usedNames := make(map[string]bool)
```

Comment:
Consider adding test coverage for usedNames in case index name collides with table name (assuming this can happen in dynamodb and that's why we added this handling).